_posts/2021-02-20-kotlin-use-site-targets.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title:  "Unwrapping Use-site Targets in Kotlin"
+date:   2021-03-02 10:00:00 +0200
+comments: true
+img:
+  href: 2021-03-02-dogs-of-berlin.png
+  copyright: Dogs of Berlin by Christian Alvart
+  alt: Dogs of Berlin
+---
+I like Kotlin for its expressiveness. Its interoperability with Java
+allows to easily use the existing Java ecosystem while enjoying a modern
+language. When it comes to annotations Kotlin handles them almost the same
+as Java. In Kotlin, a single syntax construction might be compiled into
+several JVM constructions. A Kotlin property would produce a JVM field,
+a getter, and optionally a setter. This is where annotation use-site targets
+are useful.
+
+{% include picture.html %}
+
+Take a look at the following code:
+
+```kotlin
+data class User(
+    @Unique
+    val name: String,
+    @Size(min = 18, max = 100)
+    val age: Int,
+)
+```
+
+Once it is compiled, the corresponding class would have a constructor,
+fields, getters, equals, hashcode, and some other language elements.
+If you need that in the compiled code annotations were generated specifically
```

Comment:
If you need that in the compiled code, annotations were generated specifically...